docs/source/05-02-customizing-plugins-coral-ui.md

```diff
@@ -0,0 +1,337 @@
+};
+
+export default RainbowIcon;
+````
+
+Most of the component is code generated by Sketch, except for the properties that we can control control with the palettes. The color of the rainbow lines will be given based on the order of the colors of the palette.
+
+We have two props for our component: `paletteType` and `palette`:
+
+`paletteType`: since we we have two palettes we've created, we can pass these directly as `colored` and `greyscale`
+
+`palette`: if we want to pass an array of colors we can do it using this property
+
+Ready! So now we have our icon. Now let's modify the our button `PrideButton.js` to use our new icon.
+
+
+```js
+import React from 'react';
+import cn from 'classnames';
+import styles from './PrideButton.css';
+import { withReaction } from 'plugin-api/beta/client/hocs';
+import RainbowIcon from './RainbowIcon';
+
+class PrideButton extends React.Component {
+  handleClick = () => {
+    const { postReaction, deleteReaction, alreadyReacted } = this.props;
+
+    if (alreadyReacted) {
+      deleteReaction();
+    } else {
+      postReaction();
+    }
+  };
+
+  render() {
+    const { alreadyReacted } = this.props;
+    return (
+      <div className={cn(styles.container, 'talk-plugin-pride-container')}>
+        <a
+          className={cn(styles.button, 'talk-plugin-pride-button')}
+          onClick={this.handleClick}
+        >
+          {alreadyReacted ? (
+            <RainbowIcon />
+          ) : (
+            <RainbowIcon paletteType="grayscale" />
+          )}
+        </a>
+      </div>
+    );
+  }
+}
+
+export default withReaction('pride')(PrideButton);
+```
+
+We will use the property `alreadyReacted` to change the icon and render one in grayscale (using the property `grayscale`).
+
+There are many pros and cons of using inline SVGs that are outside the scope of this tutorial. If you'd like to learn more, you can read [5 Gotchas You're Gonna Face Getting Inline SVG Into Production](https://css-tricks.com/gotchas-on-getting-svg-into-production/) and its follow-up post [Part 2 Gotchas](https://css-tricks.com/gotchas-getting-inline-svg-production-part-ii/).
+
+You can view the source code up to this point here: [talk-plugin-pride @ ae5c1a5](https://github.com/coralproject/talk-plugin-pride/commit/ae5c1a5e26390b9374c87ce5530d60c10b5c325e).
+
+To keep performance top of mine, and given that this portion of SVG code can not be cached, we will create separate SVG files for the two icon states.
+
+We will create the folder `assets` and place our two files inside it: `ColoredRainbowIcon.svg` and `GrayscaleRainbowIcon.svg`. We can export them both with Sketch or simply copy the SVG code into each file.
+
+## Using an SVG in our components
+
+We are going to import our SVG icons just as we did with our components, the only difference is the `.svg` at the end.
+
+```js
+import ColoredRainbowIcon from '../assets/ColoredRainbowIcon.svg';
+import GrayscaleRainbowIcon from '../assets/GrayscaleRainbowIcon.svg’;
```

Comment:
Looks like the quote at the end is wrong here.

Reply:
Fixed!